babel/core.py

```diff
@@ -1190,22 +1233,23 @@ def parse_locale(identifier: str, sep: str = '_') -> tuple[str, str | None, str
     if parts:
         raise ValueError(f"{identifier!r} is not a valid locale identifier")
 
-    return lang, territory, script, variant
+    return lang, territory, script, variant, modifier
 
 
-def get_locale_identifier(tup: tuple[str, str | None, str | None, str | None], sep: str = '_') -> str:
+def get_locale_identifier(tup: tuple[str, str | None, str | None, str | None, str | None], sep: str = '_') -> str:
```

Comment:
This function should also support 4-tuples. In fact, it would be best if it supported any tuple size from 1 to 5. (I think the code already does, but the signature doesn't reflect that.)

Reply:
Hm, I am not aware of a way to represent this in the function signature. Yes, I think the code does support it already, but I am unsure how else you want the signature to look. Any further hints?

Reply:
I think it'll be a bit unwieldy, but...

```suggestion
def get_locale_identifier(
    tup: tuple[str]
    | tuple[str, str | None]
    | tuple[str, str | None, str | None]
    | tuple[str, str | None, str | None, str | None]
    | tuple[str, str | None, str | None, str | None, str | None],
    sep: str = "_",
) -> str:
```